base/src/test/java/de/n26/n26androidsamples/base/data/store/MemoryReactiveStoreTest.java

```diff
@@ -46,43 +48,53 @@ public void lastStoredObjectIsEmittedAfterSubscription() {
         reactiveStore.getSingular("ID1").test().assertValue(ofObj(model));
     }
 
+    @SuppressWarnings("unchecked")
     @Test
     public void streamsEmitWhenSingleObjectIsStored() {
         List<TestObject> list = createTestObjectList();
         TestObject model = new TestObject("ID1");
-        new ArrangeBuilder().withCachedObjectList(list)
-                            .withCachedObject(model);
+        new ArrangeBuilder().withEmptyCache();
+        TestObserver<Option<TestObject>> singleTestObserver = reactiveStore.getSingular("ID1").test();
+        TestObserver<Option<List<TestObject>>> listTestObserver = reactiveStore.getAll().test();
+        new ArrangeBuilder().withCachedObjectList(list);
 
         reactiveStore.storeSingular(model);
 
-        reactiveStore.getSingular("ID1").test().assertValue(ofObj(model));
-        reactiveStore.getAll().test().assertValue(ofObj(list));
+        singleTestObserver.assertValues(none(), ofObj(model));
+        listTestObserver.assertValues(none(), ofObj(list));
```

Comment:
This is a good catch! Thanks for fixing it :)

I think the test is now a little bit messy and makes sense to separate it in two: 
````
    @SuppressWarnings("unchecked")
    @Test
    public void singularStreamEmitsWhenSingleObjectIsStored() {
        TestObject model = new TestObject("ID1");
        new ArrangeBuilder().withEmptyCache();

        TestObserver<Option<TestObject>> to = reactiveStore.getSingular("ID1").test();
        reactiveStore.storeSingular(model);

        to.assertValues(none(), ofObj(model));
    }
`````
AND
`````
    @SuppressWarnings("unchecked")
    @Test
    public void allStreamEmitsWhenSingleObjectIsStored() {
        List<TestObject> list = createTestObjectList();
        new ArrangeBuilder().withCachedObjectList(list);

        TestObserver<Option<List<TestObject>>> to = reactiveStore.getAll().test();
        reactiveStore.storeSingular(new TestObject(""));

        Mockito.verify(cache, Mockito.times(2)).getAll();
        to.assertValues(ofObj(list), ofObj(list));
    }
`````

Same for the other two tests affected. 
I think we can do two things for this PR:
1. If you are willing to put a bit more of time on this, separate the tests in two following my suggestion.
2. In the case you don't want to spend more time on this then undo the last commit, merge the fix for the `UnwrapOptionTransformerTest.java` and I will fix and separate this tests.

Let me know what you prefer.

Reply:
I will go with the first option, and separate the tests.

Reply:
Yes, makes sense. Thanks for taking the time :)

Reply:
It's done.